pkg/controller/realmadmin/show.go

```diff
@@ -66,7 +68,45 @@ func (c *Controller) HandleShow() http.Handler {
 			return
 		}
 
-		c.renderShow(ctx, w, realm, stats, userStats)
+		// If the user's requested CSV, return it as such.
+		if _, wantCSV := r.URL.Query()["csv"]; wantCSV {
+			wr := csv.NewWriter(w)
```

Comment:
Need to defer flushing this:

```go
defer wr.Flush()
```

Reply:
I called it explicitly below, but I guess since the csv writer's the only thing that can throw an error, your approach is better.